### PR TITLE
begin resister-color-duo ts mentoring notes

### PR DESCRIPTION
**Pull request by JimLynchCodes**
This PR creates the initial mentoring doc for the "colors-duo-resister" problem. i'm open to suggestions for adding changing to this doc so give me some comments please! 🤓

### CODE REVIEW

File: tracks/typescript/exercises/resister-color-duo/mentoring.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+  }
+
+}
+```
+
+### Common suggestions
+
+- The student should create an array containing the list of color strings. This should be `private` and `readonly`.
+
+- Look for an opportunity to put the access modifier `private` keyword in the contructor for the shorthand way of assigning it to an instance variable.
```
Comment (JesseSingleton) [inline tracks/typescript/exercises/resister-color-duo/mentoring.md:39]: Small typo here - should be `Constructor`!

File: tracks/typescript/exercises/resister-color-duo/mentoring.md [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+- Ensure the student checks for the error case early on as to short-circuit the program as quickly as possible.
+
+- If the student hardcodes numbers (such as _2)_, point out that rather than having this "magic number" it could be a constant or readonly instance variable.
+
+- Ensure the student uses proper type annotations, but consider suggesting they omit the types in "obvious" cases (such as when immediately assigning the variable to concrete value).
+
+- Any final comments on chosen names, formatting, unncessary code / comments / console.log's
```
Comment (JesseSingleton) [inline tracks/typescript/exercises/resister-color-duo/mentoring.md:47]: Small typo here - should be `unnecessary`.

File: tracks/typescript/exercises/resister-color-duo/mentoring.md [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    
+    const result = this.colors
+      .slice(0, this.MINIMUM_ACCEPTABLE_COLORS)
+      .reduce(
+        (accumulator: string, current: string) =>
+          `${accumulator}${this.Colors.indexOf(current)}`,
+          ''
+      );
+
+    return parseInt(result);
```
Comment (JesseSingleton) [inline tracks/typescript/exercises/resister-color-duo/mentoring.md:28]: I've noticed that across the JS track at least we seem to prefer `Number()` over `parseX`, with the only reasoning I've ever found being that "it's more declarative". Thoughts on this? Are both acceptable?

File: tracks/typescript/exercises/resister-color-duo/mentoring.md [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+    if (colors.length < this.MINIMUM_ACCEPTABLE_COLORS)
+      throw "At least two colors need to be present";
+
+  }
+
+  value = (): number => {
+    
+    const result = this.colors
+      .slice(0, this.MINIMUM_ACCEPTABLE_COLORS)
+      .reduce(
```
Comment (JesseSingleton) [inline tracks/typescript/exercises/resister-color-duo/mentoring.md:22]: This is just my opinion but I prefer a solution using `map`. i,e 
```
const result = this.colors
    .slice(0, this.MINIMUM_ACCEPTABLE_COLORS)
    .map((c) => this.Colors.indexOf(c))
    .join('')
```
I don't think using `reduce` is any less correct, it just it feels a bit of a strange use case for it here!

Happy to discuss!

**Comment by JesseSingleton** [comment]
Hey @JimLynchCodes added a few small comments, would love to hear your opinion! Nicely done on the PR!

File: tracks/typescript/exercises/resister-color-duo/mentoring.md (hunk 1 above)
Comment (JimLynchCodes) [reply tracks/typescript/exercises/resister-color-duo/mentoring.md]: fixed!

File: tracks/typescript/exercises/resister-color-duo/mentoring.md (hunk 2 above)
Comment (JimLynchCodes) [reply tracks/typescript/exercises/resister-color-duo/mentoring.md]: fixed!

File: tracks/typescript/exercises/resister-color-duo/mentoring.md (hunk 3 above)
Comment (JimLynchCodes) [reply tracks/typescript/exercises/resister-color-duo/mentoring.md]: True, I think it is just that Number is faster?

Anyway, I have changed the example code to use Number now and added a bullet point about this to the list of suggestions. 👍

File: tracks/typescript/exercises/resister-color-duo/mentoring.md (hunk 4 above)
Comment (JimLynchCodes) [reply tracks/typescript/exercises/resister-color-duo/mentoring.md]: This works and is arguably much easier to read so I changed the example to do it this way. 👍

**Comment by JimLynchCodes** [comment]
> Hey @JimLynchCodes added a few small comments, would love to hear your opinion! Nicely done on the PR!

Thanks for the review :)

File: tracks/typescript/exercises/resister-color-duo/mentoring.md (hunk 3 above)
Comment (JesseSingleton) [reply tracks/typescript/exercises/resister-color-duo/mentoring.md]: I'm not entirely sure which is faster to be honest - and on exercism I think generally performance is not something that comes up too often in reviews. I typically refer people to this [SO Post](https://stackoverflow.com/a/4090554/9758007) that goes into more detail!

**Comment by JesseSingleton** [comment]
Apologies for the delay! One minor point is that there is a typo in the file path tracks/typescript/exercises/resister-color-duo/mentoring.md should be tracks/typescript/exercises/**resistor-color-duo**/mentoring.md.

Else this looks reasonable to me. Unfortunately after starting this review I realised that contributors don't count as an approving review! So you'll need one of the maintainers to take a look too. Sorry! But looks good!

File: tracks/typescript/exercises/resister-color-duo/mentoring.md [hunk 5]
```diff
@@ -0,0 +1,45 @@
+### Reasonable Solutions
```
Comment (SleeplessByte) [inline tracks/typescript/exercises/resister-color-duo/mentoring.md:1]: Have a look the JavaScript mentor notes which has a more idiomatic approach and _more_ allowed variations.

**Comment by SleeplessByte** [comment]
@JimLynchCodes if you're still interested; these notes look pretty good with a few minor things to change.

**Review by SleeplessByte — APPROVED** [review]

**Pull request closed without merging**